### PR TITLE
[FLINK-23961][kubernetes] Hint user about missing 'kubernetes.cluster…

### PR DESCRIPTION
**Pull request by dmvk**
https://issues.apache.org/jira/browse/FLINK-23961

### CODE REVIEW

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java [hunk 1]
```diff
@@ -85,7 +85,15 @@ public Fabric8FlinkKubeClient(
             Configuration flinkConfig,
             NamespacedKubernetesClient client,
             ExecutorService executorService) {
-        this.clusterId = checkNotNull(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID));
+        this.clusterId =
```
Comment (Aitozi) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java:88]: why not just use `checkNotNull(flinkConfig.getString(KubernetesConfigOptions.CLUSTER_ID), String.format(
                                                        "Configuration option '%s' is not set.",
                                                        KubernetesConfigOptions.CLUSTER_ID.key())` ?

Comment (dmvk) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java:88]: `NullPointerException` didn't really feel suitable here as this is basically a misconfiguration on the user side.

NPE is more suitable for API misuse (something that can be basically avoided doing static code analysis).

Comment (Aitozi) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java:88]: Get it. +1 for this.

**Comment by dmvk** [comment]
Failed with FLINK-22366

**Review by XComp — APPROVED** [review]
LGTM 👍

**Comment by dmvk** [comment]
@flinkbot run azure

**Pull request merged**